xtask/src/ci.rs

```diff
@@ -241,6 +260,70 @@ fn run_wasm_checks(cmd: Option<WasmFeatureSet>) -> Result<()> {
     Ok(())
 }
 
+fn run_wasm_pack_tests(cmd: Option<WasmFeatureSet>) -> Result<()> {
+    let args = BTreeMap::from([
+        (WasmFeatureSet::MatrixQrcode, ("matrix-qrcode", "")),
+        (
+            WasmFeatureSet::MatrixSdkNoDefault, ("matrix-sdk", "--no-default-features --features rustls-tls --lib")
+        ),
+        (WasmFeatureSet::MatrixSdkBase, ("matrix-sdk-base", "")),
+        (WasmFeatureSet::MatrixSdkCommon, ("matrix-sdk-common", "")),
+        (WasmFeatureSet::MatrixSdkCrypto, ("matrix-sdk-crypto", "")),
+        (
+            WasmFeatureSet::MatrixSdkIndexeddbStoresNoCrypto, (
+                "matrix-sdk", "--no-default-features --features indexeddb-state-store,rustls-tls",
+            )
+        ),
+        (
+            WasmFeatureSet::MatrixSdkIndexeddbStores, (
+                "matrix-sdk", "--no-default-features --features indexeddb-state-store,indexeddb-crypto-store,encryption,rustls-tls --lib",
+            )
+        ),
+        (
+            WasmFeatureSet::IndexeddbNoCrypto, (
+                "matrix-sdk-indexeddb", "--no-default-features --features state-store",
+            )
+        ),
+        (
+            WasmFeatureSet::IndexeddbWithCrypto, (
+                "matrix-sdk-indexeddb", "--no-default-features --features state-store,crypto-store",
+            )
+        ),
+    ]);
+
+    let run = |(folder, arg_set): (&str, &str)| {
```

Comment:
Does `wasm-pack` not support `-p`, or why do we have this separate `folder` thing instead of `-p` flags?

Reply:
that's correct,`wasm-pack` is not a cargo-related system but has its own, more JS-worldy, style of doing things. That comes with two disadvantages: 1) it doesn't know about cargo workspaces (though you can pass through `-p` on the cargo side, it does some look ups itself and that often fails if you are in the wrong directory); 2) it only knows about the local tests (everything else is stripped from the final wasm-binary).

As a result, we to test various components, we have to navigate into their directory and run `wasm-pack` from there for reliable results.